yt_dlp/extractor/startrek.py

```diff
@@ -0,0 +1,51 @@
+from .common import InfoExtractor
+from ..utils import int_or_none, urljoin
+
+
+class StarTrekIE(InfoExtractor):
+    _VALID_URL = r'(?P<base>https://(?:intl|www)\.startrek\.com)/videos/(?P<id>[^/]+)'
+    _TESTS = [{
+        'url': 'https://intl.startrek.com/videos/watch-welcoming-jess-bush-to-the-ready-room',
+        'md5': '491df5035c9d4dc7f63c79caaf9c839e',
+        'info_dict': {
+            'id': 'watch-welcoming-jess-bush-to-the-ready-room',
+            'ext': 'mp4',
+            'title': 'WATCH: Welcoming Jess Bush to The Ready Room',
+            'duration': 1888,
+            'timestamp': 1655388000,
+            'upload_date': '20220616',
+            'description': 'md5:1ffee884e3920afbdd6dd04e926a1221',
+            # Thumbnail seems to have some tracking parameters added, ignore those.
+            'thumbnail': r're:https://intl\.startrek\.com/sites/default/files/styles/video_1920x1080/public/images/2022-06/pp_14794_rr_thumb_107_yt_16x9.jpg(?:\?.+)?',
+        }
+    }]
+
+    def _real_extract(self, url):
+        urlbase, video_id = self._match_valid_url(url).group('base', 'id')
+        webpage = self._download_webpage(url, video_id)
+
+        player = self._search_regex(
+            r'(<\s*div\s+id\s*=\s*"cvp-player-[^<]+<\s*/div\s*>)', webpage, 'player')
+
+        hls = self._html_search_regex(r'\bdata-hls\s*=\s*"([^"]+)"', player, 'HLS URL')
+        formats, subtitles = self._extract_m3u8_formats_and_subtitles(hls, video_id)
+        self._sort_formats(formats)
+
+        json_ld = self._search_json_ld(webpage, video_id, fatal=False)
+
+        return {
+            'id': video_id,
+            'title': self._html_search_regex(
+                r'\bdata-title\s*=\s*"([^"]+)"', player, 'title', json_ld.get('title')),
+            'description': self._html_search_regex(
+                r'(?s)<\s*div\s+class\s*=\s*"header-body"\s*>(.+?)<\s*/div\s*>',
+                webpage, 'description', fatal=False),
+            'duration': int_or_none(self._html_search_regex(
+                r'\bdata-duration\s*=\s*"(\d+)"', player, 'duration', fatal=False)),
```

Comment:
not really a burning issue but these are ugly af, I would rather not have these...but upto u if u want to keep them or not

Reply:
wdym "not have these"?

Reply:
I personally would have not parsed them and not included them in the infodict as these 2 are not really useful imo. 

I would say 90% of the users never only use dlp as `yt-dlp url` and these 2 fields are not used in default flow afaik. And since this extractor is not a request (OP is adding it proactively), it is okay to have cleaner code than perfect code imo.

But as i said, this is not something i feel very strong about. I just found them ugly and hence pointed them out.

Reply:
Well, why throw away perfectly good metadata?

And to be honest, I found the code here more readable before pukkandan explicitly asked me to [make the regexes more resilient](https://github.com/yt-dlp/yt-dlp/pull/4191#discussion_r906815237) and [inline the expressions](https://github.com/yt-dlp/yt-dlp/pull/4191#discussion_r906815423) ;) But that’s a matter of preference I guess.

Reply:
yeah definitely upto owner and/or developer. We all have our own coding style and preferences anyways :)